Assets/Scripts/Round/RoundManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections;
+using Mirror;
+using TMPro;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Round
+{
+    /// <summary>
+    /// Behaviour responsible for syncing timers between server and clients.
+    /// Also handles starting and restarting rounds.
+    /// Should be attached to the RoundManager prefab.
+    /// </summary>
+    public class RoundManager : NetworkBehaviour
+    {
+        //How long will the warmup period be. Starts immediately on server start.
+        [SerializeField] private int warmupTimeSeconds = 30;
+        //Text element that stores the timer
+        [SerializeField] private TextMeshProUGUI timerText = null;
+        //UI element that contains the round timer
+        [SerializeField] private RectTransform timerUi = null;
+        //UI element that contains the round controls
+        [SerializeField] private RectTransform controlUi = null;
+
+        private int timerSeconds = 0;
+        private bool started = false;
+        private Coroutine tickCoroutine;
+            
+        public bool IsRoundStarted => started;
+
+        public void StartWarmup()
+        {
+            timerSeconds = warmupTimeSeconds;
+            StartCoroutine(TickWarmup());
+        }
+
+        public void StartRound()
+        {
+            started = true;
+            controlUi.gameObject.SetActive(true);
+            tickCoroutine = StartCoroutine(Tick());
+        }
+        
+        public void RestartRound()
+        {
+            if (!isServer) return;
+            
+            StopCoroutine(tickCoroutine);
+            NetworkManager.singleton.ServerChangeScene(SceneManager.GetActiveScene().name);
+        }
+        
+        private IEnumerator TickWarmup()
+        {
+            while (timerSeconds > 0)
+            {
+                RpcUpdateClientClocks(GetTimerText());
```

Comment:
Could we reduce the number of web requests by telling the client the start time, then letting the client count the seconds independently?

Reply:
If the client freezes or otherwise desyncs that would cause issues. My reasoning was that given how often things like network transforms are updated, one extra call per second is trivial to guarantee perfect sync.

Reply:
Seems fine then. I think if anything sending only one request (the server sends the time the countdown started at and how many seconds total) would result in less desync issues but yeah the number of requests is still pretty minimal.

Also i have the feeling that this code will be revisited eventually when server states start getting more complicated anyway.